pandas/core/indexes/multi.py

```diff
@@ -3614,7 +3614,7 @@ def intersection(self, other, sort=False):
         if self.equals(other):
             if self.has_duplicates:
                 return self.unique().rename(result_names)
-            return self._get_reconciled_name_object(other)
```

Comment:
is this not equivalent?

Reply:
No, my test returned c,b as names with the other version

Reply:
```
mi = MultiIndex.from_arrays([[1, 2, 1], [3, 4, 2]], names=["c", "b"])
mi2 = MultiIndex.from_arrays([[1, 2, 1], [3, 4, 2]], names=["a", "b"])

print(mi.intersection(mi2))
```

returns

```
MultiIndex([(1, 3),
            (2, 4),
            (1, 2)],
           names=['c', 'b'])
```

Output on master

Reply:
is this indicative of another problem?  could we align the behavior by overriding MultiIndex._get_reconcile_name_object?

Reply:
Issue is deeper down the stack, we would have to change ``_maybe_match_name``, but this is doable. Would have to override the method then in multi

Reply:
makes sense, thanks.  eventually we should try to de-duplicate this further, but im happy with this PR as is.

Reply:
Other PR would help here.